features/telemetry/protocols/netflow/adapter/src/test/java/org/opennms/netmgt/telemetry/protocols/netflow/adapter/netflow5/Netflow5ConverterTest.java

```diff
@@ -121,9 +121,15 @@ public void canParseNetflow5Flows() {
                 header = new Header(slice(buffer, Header.SIZE));
                 final Packet packet = new Packet(header, buffer);
                 packet.getRecords().forEach(rec -> {
-                    final ByteBuffer bf = ParserBase.serialize(Protocol.NETFLOW9, rec);
-                    final BsonDocument doc = new RawBsonDocument(bf.array());
-                    flows.addAll(nf5Converter.convert(doc));
+
+                    byte[] message = FlowMessageBuilderFactory.buildAndSerialize(Protocol.NETFLOW5, rec);
+                    try {
+                        FlowMessage flowMessage = FlowMessageProto.parseFrom(message);
+                        flows.addAll(nf5Converter.convert(flowMessage));
+                    } catch (InvalidProtocolBufferException e) {
```

Comment:
Just make the method `throws Exception`.

Reply:
It's in lambda expression

Reply:
Right. But in the end it could be a good old for loop, right?

Reply:
Since `packet.getRecords()` is a stream, I'm inclined to keep this way.